service/src/test/java/bio/terra/externalcreds/dataAccess/SshKeyPairInternalDaoTest.java

```diff
@@ -84,6 +121,88 @@ void testGetSshKeyPair() throws NoSuchAlgorithmException, IOException {
+      try (var utilsMock = Mockito.mockStatic(EncryptDecryptUtils.class)) {
+        utilsMock
+            .when(
+                () ->
+                    EncryptDecryptUtils.encryptSymmetric(
+                        eq(externalCredsConfig.getServiceGoogleProject()),
+                        eq(location),
+                        eq(keyRing),
+                        eq(encryptionKey),
+                        eq(sshKey.getPrivateKey())))
+            .thenReturn(cypheredkey);
+        utilsMock
+            .when(
+                () ->
+                    EncryptDecryptUtils.decryptSymmetric(
+                        eq(externalCredsConfig.getServiceGoogleProject()),
+                        eq(location),
+                        eq(keyRing),
+                        eq(encryptionKey),
+                        eq(cypheredkey)))
+            .thenReturn(sshKey.getPrivateKey());
+
+        sshKeyPairDAO.upsertSshKeyPair(sshKey);
+
+        var loadedSshKeyOptional =
+            sshKeyPairDAO.getSshKeyPair(sshKey.getUserId(), sshKey.getType());
+
+        assertPresent(loadedSshKeyOptional);
+        verifySshKeyPair(sshKey, loadedSshKeyOptional.get());
+      }
+    }
+
+    @Test
+    void decryptKeyFails() throws NoSuchAlgorithmException, IOException {
+      var sshKey = createRandomGithubSshKey();
+      when(externalCredsConfig.getEnableKmsEncryption()).thenReturn(true);
+      var cypheredkey = "jfidosruewr1k=";
+      String location = "us-central1";
+      when(externalCredsConfig.getKeyRingLocation()).thenReturn(Optional.of(location));
+      String keyRing = "key_ring";
+      when(externalCredsConfig.getKeyRingId()).thenReturn(Optional.of(keyRing));
+      String encryptionKey = "encryption_key";
+      when(externalCredsConfig.getKeyId()).thenReturn(Optional.of(encryptionKey));
+      try (var utilsMock = Mockito.mockStatic(EncryptDecryptUtils.class)) {
+        utilsMock
+            .when(
+                () ->
+                    EncryptDecryptUtils.encryptSymmetric(
+                        eq(externalCredsConfig.getServiceGoogleProject()),
+                        eq(location),
+                        eq(keyRing),
+                        eq(encryptionKey),
+                        eq(sshKey.getPrivateKey())))
+            .thenReturn(cypheredkey);
+        utilsMock
+            .when(
+                () ->
+                    EncryptDecryptUtils.decryptSymmetric(
+                        eq(externalCredsConfig.getServiceGoogleProject()),
+                        eq(location),
+                        eq(keyRing),
+                        eq(encryptionKey),
+                        eq(cypheredkey)))
+            .thenThrow(new IOException("Something went wrong, decryption fails"));
+        sshKeyPairDAO.upsertSshKeyPair(sshKey);
+        assertThrows(
+            ExternalCredsException.class,
+            () -> sshKeyPairDAO.getSshKeyPair(sshKey.getUserId(), sshKey.getType()));
```

Comment:
Please see SonarCloud feedback about handling RuntimeExceptions from this lambda and a similar one also: https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=122&id=DataBiosphere_terra-external-credentials-manager&open=AX_12dPla-RX1g1tBSPF

Reply:
obsolete as this test is deleted.